pkg/types/machinepools.go

```diff
@@ -125,3 +132,29 @@ func (p *MachinePoolPlatform) Name() string {
 		return ""
 	}
 }
+
+// WorkloadName is the name of the workload
+// +kubebuilder:validation:Enum="";management
+type WorkloadName string
+
+const (
+	// ManagementWorkload pins all infrastructure pods to the defined
+	// cpuset.
+	ManagementWorkload WorkloadName = "management"
+)
+
+// Workload defines a set of CPUs with a name, which will be used by Kubelet
+// and Cri-O to partition specific sets of pods to those CPUs exclusively.
+type Workload struct {
+	// The name of the workload
+	Name WorkloadName `json:"name"`
+
+	// CPUIds is a CPU set specifier for the CPUs to add to the isolated
+	// set for this workload, using the same cpuset syntax used elsewhere
+	// in kubernetes. Asking for explicit IDs instead of simply a count
+	// gives the user control in situations where they need specific CPUs
+	// to be available because they're close to accelerators, NICs, or
+	// other special hardware.  For example, "1,5,9" or "0-4" or
+	// "0-1,52-53" +kubebuilder:validation:Pattern=[0-9][-,0-9]*
```

Comment:
```suggestion
	// "0-1,52-53"
	// +kubebuilder:validation:Pattern=[0-9][-,0-9]*
```

Reply:
Good catch, not sure how that formatting got broken.